actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -94,9 +99,11 @@ module ClassMethods
       # * <tt>:reset_session</tt> - Resets the session.
       # * <tt>:null_session</tt> - Provides an empty session during request but doesn't reset it completely. Used as default if <tt>:with</tt> option is not specified.
       def protect_from_forgery(options = {})
+        options.reverse_merge!(prepend: true)
```

Comment:
And where? The idea behind this is to make `prepend: true` default (to keep backwards compatibility), but be able to pass `prepend: false` to `before_action` when you need.